pkg/cmd/server/crypto/crypto.go

```diff
@@ -113,14 +114,21 @@ type CA struct {
 	SerialFile string
 	Serial     int64
 	Config     *TLSCertificateConfig
+	Output     io.Writer
 }
 
-func EnsureCA(certFile, keyFile, serialFile, name string) (*CA, error) {
+func (ca *CA) Printf(tmpl string, args ...interface{}) {
+	if ca.Output != nil {
+		fmt.Fprintf(ca.Output, tmpl, args...)
+	}
+}
+
+func EnsureCA(certFile, keyFile, serialFile, name string, out io.Writer) (*CA, error) {
```

Comment:
Return a "created" bool here and let the caller decide whether to report it or not.